spec/coerce_spec.rb

```diff
@@ -188,6 +188,22 @@ class Param2 < T::Struct
     end
   end
 
+  context 'when dealing with hashes'  do
```

Comment:
can we add more test cases? at least there should be a case that the coercion of a key or a value failed

Reply:
Added an invalid value case. Since we are handling the inner types recursively, a lot of the test cases will be similar to the existing ones.